enumfields/fields.py

```diff
@@ -31,7 +32,38 @@ def __set__(self, obj, value):
         obj.__dict__[self.field.name] = self.field.to_python(value)
 
 
-class EnumFieldMixin:
+class EnumFieldValidationMixin:
+
+    def _raise_error_if_next_value_not_allowed(self, value, model_instance):
+        if model_instance.pk is not None:
+            previous_values = [choice.value for choice in self.enum.get_allowed_previous_choices(value.name)]
+
+            # search whether in db exists record within all posible previous states and the current one
+            if not self.model.objects.filter(
+                Q(**{'{}__in'.format(self.name): previous_values + [value.value]}) | Q(**{self.name: None}),
+                pk=model_instance.pk,
+            ).exists():
+                raise ValidationError(
+                    ugettext(
+                        'Transition from current choice to {} choice is not allowed').format(
+                        value.name)
+                )
+
+    def _raise_error_if_initial_value_not_allowed(self, value, model_instance):
+        if model_instance.id is None:
+            initial_choices = self.enum.get_initial_choices()
+            if value not in initial_choices:
```

Comment:
Why don't you just read `inital` attribute?

Reply:
right, I have overlooked that.